### PR TITLE
Housekeeping: Remove redundant stuff from Go code

### PR DESCRIPTION
**Pull request by vdavid**
This is a refactor that is meant to improve code quality without changing behavior.

Four changes:
- Removed redundant type definitions
- Removed redundant import aliases
- Removed redundant type conversions
- Removed redundant parentheses

Tip for reviewers: review this PR commit-by-commit for clarity! Here is a 1-minute video for guidance on the commits: https://www.loom.com/share/571167a3eaec460fbbf22b908d4f8354

## Test plan

CI should pass

### CODE REVIEW

File: lib/codeintel/lsif/scip/document_test.go [hunk 1]
```diff
@@ -75,8 +75,8 @@ func TestConvertLSIFDocument(t *testing.T) {
 	}
 
 	targetRangeFetcher := func(resultID precise.ID) []precise.ID {
-		if resultID == precise.ID("d0") {
-			return []precise.ID{precise.ID("r0")}
+		if resultID == ("d0") {
```
Comment (efritz) [inline lib/codeintel/lsif/scip/document_test.go:78]:
```suggestion
		if resultID == "d0" {
```

**Comment by camdencheek** [comment]
Shouldn't the unnecessary `string()` conversions be caught by the `unconvert` lint?

File: enterprise/internal/codeintel/shared/types/scip_sanitize.go [hunk 2]
```diff
@@ -97,6 +97,6 @@ func sanitizeString(s string) string {
 		return s
 	}
 
-	// magic
+	//goland:noinspection GoRedundantConversion – this seems redundant, but it magically makes the string valid UTF-8
```
Comment (efritz) [inline enterprise/internal/codeintel/shared/types/scip_sanitize.go:100]: <3

**Review by efritz — APPROVED** [review]

File: enterprise/internal/codeintel/shared/types/scip_sanitize.go [hunk 3]
```diff
@@ -97,6 +97,6 @@ func sanitizeString(s string) string {
...
 	return string([]rune(s))
```
Comment (camdencheek) [inline enterprise/internal/codeintel/shared/types/scip_sanitize.go:101]: This is just ignoring a lint, yes?

File: enterprise/internal/codeintel/shared/types/scip_sanitize.go (hunk 2 above)
Comment (camdencheek) [inline enterprise/internal/codeintel/shared/types/scip_sanitize.go:100]: I'm a little hesitant to make a pattern of editor-specific lint pragmas 🤔

**Review by camdencheek — APPROVED** [review]

File: cmd/frontend/graphqlbackend/executor.go [hunk 4]
```diff
@@ -58,7 +58,7 @@ func (e *ExecutorResolver) Compatibility() (*string, error) {
 }
 
 func calculateExecutorCompatibility(ev string) (*string, error) {
-	var compatibility ExecutorCompatibility = ExecutorCompatibilityUpToDate
+	var compatibility = ExecutorCompatibilityUpToDate
```
Comment (Strum355) [inline cmd/frontend/graphqlbackend/executor.go:61]:
```suggestion
	compatibility := ExecutorCompatibilityUpToDate
```

File: dev/sg/internal/bk/bk.go [hunk 5]
```diff
@@ -183,7 +183,7 @@ func (c *Client) GetJobAnnotationsByBuildNumber(ctx context.Context, pipeline st
 		return nil, err
 	}
 
-	var result JobAnnotations = make(JobAnnotations, 0)
+	var result = make(JobAnnotations, 0)
```
Comment (Strum355) [inline dev/sg/internal/bk/bk.go:186]:
```suggestion
	result := make(JobAnnotations, 0)
```

File: enterprise/internal/insights/pipeline/backfill.go [hunk 6]
```diff
@@ -154,9 +154,9 @@ func makeSearchJobsFunc(logger log.Logger, commitClient GitCommitClient, compres
 			return &reqContext, jobs, err
 		}
 		searchPlan := compressionPlan.Filter(ctx, req.SampleTimes, req.Repo.Name)
-		var ratio float64 = 1.0
+		var ratio = 1.0
```
Comment (Strum355) [inline enterprise/internal/insights/pipeline/backfill.go:157]:
```suggestion
		ratio := 1.0
```

File: enterprise/internal/insights/resolvers/insight_view_resolvers.go [hunk 7]
```diff
@@ -1462,8 +1462,8 @@ func filtersFromInput(input *graphqlbackend.InsightViewFiltersInput) types.Insig
 }
 
 func sortSeriesResolvers(ctx context.Context, seriesOptions types.SeriesDisplayOptions, resolvers []graphqlbackend.InsightSeriesResolver) ([]graphqlbackend.InsightSeriesResolver, error) {
-	var sortMode types.SeriesSortMode = types.ResultCount
-	var sortDirection types.SeriesSortDirection = types.Desc
+	var sortMode = types.ResultCount
+	var sortDirection = types.Desc
 	var limit int32 = 20
```
Comment (Strum355) [inline enterprise/internal/insights/resolvers/insight_view_resolvers.go:1467]: seems to be a lot of these, not gonna point out all of em :P can change them if you want, or ignore me

File: enterprise/internal/codeintel/shared/types/scip_sanitize.go (hunk 2 above)
Comment (vdavid) [reply enterprise/internal/codeintel/shared/types/scip_sanitize.go:100]: Noted, removed it.

File: enterprise/internal/insights/resolvers/insight_view_resolvers.go (hunk 7 above)
Comment (vdavid) [reply enterprise/internal/insights/resolvers/insight_view_resolvers.go]: I went through my code and fixed it in all my changes. Not sure if we strongly prefer short var declarations, but they are definitely more concise.

**Pull request merged**